pkg/sidecarproxy/metricshandler/jupyterkernelbusyness/metricshandler.go

```diff
@@ -0,0 +1,162 @@
+package jupyterkernelbusyness
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/v3io/sidecar-proxy/pkg/sidecarproxy/metricshandler"
+	"github.com/v3io/sidecar-proxy/pkg/sidecarproxy/metricshandler/abstract"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type metricsHandler struct {
+	*abstract.MetricsHandler
+	metric *prometheus.GaugeVec
+}
+
+func NewMetricsHandler(logger logger.Logger,
+	forwardAddress string,
+	listenAddress string,
+	namespace string,
+	serviceName string,
+	instanceName string) (metricshandler.MetricHandler, error) {
+
+	newJupyterKernelBusynessMetricsHandler := metricsHandler{}
+	newAbstractMetricsHandler, err := abstract.NewMetricsHandler(
+		logger.GetChild(string(metricshandler.JupyterKernelBusynessMetricName)),
+		forwardAddress,
+		listenAddress,
+		namespace,
+		serviceName,
+		instanceName,
+		metricshandler.JupyterKernelBusynessMetricName)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create abstract metric handler")
+	}
+
+	newJupyterKernelBusynessMetricsHandler.MetricsHandler = newAbstractMetricsHandler
+
+	return &newJupyterKernelBusynessMetricsHandler, nil
+}
+
+func (n *metricsHandler) RegisterMetrics() error {
+	gaugeVec := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: string(n.MetricName),
+		Help: "Jupyter kernel busyness",
+	}, []string{"namespace", "service_name", "instance_name"})
+
+	if err := prometheus.Register(gaugeVec); err != nil {
+		return errors.Wrapf(err, "Failed to register metric: %s", string(n.MetricName))
+	}
+
+	n.Logger.InfoWith("Metric registered successfully", "metricName", string(n.MetricName))
+	n.metric = gaugeVec
+
+	return nil
+}
+
+func (n *metricsHandler) Start() {
+	n.Logger.Info("Starting jupyter kernel busyness metrics handler")
+	ticker := time.NewTicker(5 * time.Second)
+	errc := make(chan error)
+	go func() {
+		for range ticker.C {
+			kernels, err := n.getKernels()
+			if err != nil {
+				errc <- errors.Wrap(err, "Failed to get kernels")
+			}
+
+			isBusyKernelExists := n.isBusyKernelExists(kernels)
+			if isBusyKernelExists {
+				if err := n.setMetric(BusyKernelExecutionState); err != nil {
+					errc <- errors.Wrapf(err, "Failed to set metric")
+				}
+			} else {
+
+				// If none of the kernels is busy - it's idle
+				if err := n.setMetric(IdleKernelExecutionState); err != nil {
+					errc <- errors.Wrapf(err, "Failed to set metric")
```

Comment:
Either determine metric value depending on business and then call `setMetric` once, or (less ideal) at least change the error message to differentiate between a failure on line 78 to line 84 (I think it's best to prepare param first and call the function once)